mono/mini/cfold.c

```diff
@@ -97,13 +97,13 @@ mono_constant_fold_ins (MonoCompile *cfg, MonoInst *ins, MonoInst *arg1, MonoIns
 	switch (ins->opcode) {
 	case OP_FADD:
 	case OP_FMUL:
-		if (arg2->opcode == OP_R8CONST && arg2->opcode == OP_R8CONST){
+		if ((arg1->opcode == OP_R8CONST || arg1->opcode == OP_R4CONST) && arg2->opcode == arg1->opcode){
 			ALLOC_DEST (cfg, dest, ins);
```

Comment:
Needs an extra space at the end.

Reply:
This won't work because OP_R4CONST has its inst_p0 field pointing to a float, not a double, so FOLD_BINOP will not work.

Reply:
Thanks for noticing it, I fixed it. I don't think I have actually ever hit this case since Roslyn already does this optimization (for non-inlined code). The main issue I saw was the cut & paste error in the condition.